corehq/form_processor/models.py

```diff
@@ -1395,6 +1395,9 @@ def __eq__(self, other):
             self.form_id == other.form_id
         )
 
+    def __hash__(self):
+        return hash((self.case_id, self.type, self.form_id))
```

Comment:
huh, can there not be multiple updates to a case in a single form?  This `__hash__` definition is consistent with `__eq__`, so it seems fine, but I didn't realize that.